docs/pages/database-access/auto-user-provisioning/postgres.mdx

```diff
@@ -90,6 +94,78 @@ Users created within the database will:
 
 (!docs/pages/includes/database-access/auto-user-provisioning/postgres15-grant-create.mdx!)
 
+### User does not have CONNECT privilege error
+
+You may encounter the following error when the admin user or the roles assigned
+to the auto-provisioned user do not have the permission to connect to the target
+database:
+
+```shell
+> tsh db connect mysql --db-name <database> example
+...
+FATAL: permission denied for database "<database>"
+DETAIL: User does not have CONNECT privilege.
+```
+
+Make sure `CONNECT` is granted to the admin user and the respective roles:
+```sql
+GRANT CONNECT ON DATABASE <database> to "teleport-admin";
+GRANT CONNECT ON DATABASE <database> to "reader";
+```
+
+### Cannot execute in a read-only transaction error
+
+You may encounter the following error when connecting to an AWS RDS Aurora
+reader endpoint:
+```shell
+> tsh db connect mysql --db-name <database> example
+psql: error: connection to server at "localhost" (::1), port 12345 failed: Connection refused
+...
+ERROR: cannot execute CREATE ROLE in a read-only transaction (SQLSTATE 25006)
+```
+
+Database auto-user provisioning is curerntly not compatible with RDS Aurora
+reader endpoints. Please use auto-user provisioning on the primary endpoints.
+
+### No schema has been selected error
+
+You may encounter the following error when connecting when the admin user does not have `USAGE`
+permission on the schema:
+```shell
+> tsh db connect mysql --db-name <database> example
+psql: error: connection to server at "localhost" (::1), port 12345 failed: Connection refused
+...
+ERROR: no schema has been selected to create in (SQLSTATE 3F000)
+```
+
+To fix this, make sure that the admin user is granted `USAGE` and `CREATE` on
+schema `public` in the target database:
+```sql
+GRANT USAGE ON SCHEMA public TO "teleport-admin";
+GRANT CREATE ON SCHEMA public TO "teleport-admin";
+```
+
+### Permission denied to grant role "rds_iam"
+
+You may encounter the following error when connecting to an RDS database:
+```shell
+> tsh db connect mysql --db-name <database> example
+psql: error: connection to server at "localhost" (::1), port 12345 failed: Connection refused
+...
+ERROR: permission denied to grant role "rds_iam" (SQLSTATE 42501)
+```
+
+This happens when the admin user does not have enough permission to grant the
+role to the auto-provisioned the user. To fix this, grant the `ADMIN` option to
```

Comment:
```suggestion
This happens when the admin user does not have permission to grant the
"rds_iam" role to other users. To fix this, grant the "rds_iam" role with the `ADMIN` option to
```

I haven't tried auto user creation with  RDS postgres - if this is always required for RDS then we should update the docs at step 1 to include the `WITH ADMIN TRUE` as well.

Reply:
This one depends on the version. It's only required in some newer versions. Though, let me check an older version works with `WITH ADMIN`. If so, i will also add it to the top.